pkgs/by-name/go/golds/info_module-gover.patch

```diff
@@ -0,0 +1,25 @@
+diff --git a/internal/server/info_module.go b/internal/server/info_module.go
+index d2513bd..ad2921e 100644
+--- a/internal/server/info_module.go
++++ b/internal/server/info_module.go
+@@ -8,6 +8,7 @@ import (
+ 	"io/ioutil"
+ 	"log"
+ 	"os"
++	"os/exec"
+ 	"path/filepath"
+ 	"regexp"
+ 	"strings"
+@@ -756,7 +757,11 @@ func findGoToolchainVersionFromGoRoot(goroot string) (string, error) {
+ 		versionData, err = ioutil.ReadFile(filepath.Join(goroot, "VERSION.cache"))
+ 	}
+ 	if err != nil {
+-		return "", fmt.Errorf("failed to get Go toolchain version in GOROOT (%s)", goroot)
++		cmd := exec.Command("go", "env", "GOVERSION")
++		versionData, err = cmd.Output()
++		if err != nil {
++			return "", fmt.Errorf("failed to get Go toolchain version in GOROOT (%s)", goroot)
++		}
```

Comment:
I don't have confidence in whether these patches are reasonable or not.
Is this patch only provided for Nix?
Don't we need to send it upstream and wait for it?

Reply:
>  Is this patch only provided for Nix?

For nix only. I've tried it with archlinux and it succeeds.

```console
# arch linux
$ cat $(go env GOROOT)/VERSION
go1.23.3
time 2024-11-06T18:46:45Z

# go from nix
$ cat $(go env GOROOT)/VERSION
cat: /nix/store/bavnchxi7v6xs077jxv7fl5rrqc3y87w-go-1.23.3/share/go/VERSION: No such file or directory

$ go env GOVERSION
go1.23.3
```

The go package has no `<goroot>/VERSION` file in the build. Maybe it was removed for reproducibility?

But `go env GOVERSION` reports the same thing

Reply:
Sure, so I understood that this patch should be required for nixpkgs.

https://github.com/NixOS/nixpkgs/blob/6357f592127ca47fc21be5e8a11ec3e7b5572637/pkgs/README.md?plain=1#L510

>only do this for patches that are unique to `nixpkgs`

:heavy_check_mark:

Reply:
1. go introduced distpack https://github.com/golang/go/commit/1635205a72d26721af54f01ccbab8e0b51ded3a9, [it ensures GOROOT/VERSION](https://github.com/golang/go/blob/b68f8ca89a2fb6b33d1e78fadc33d1c35693f6b8/src/cmd/distpack/test.go#L43)
2. nixpkgs introduced it https://github.com/NixOS/nixpkgs/pull/254497
3. nixpkgs reverted it in https://github.com/NixOS/nixpkgs/pull/256588
4. nixpkgs have revised PR for that https://github.com/NixOS/nixpkgs/pull/256505 with draft state

Reply:
Added a comment for the reason behind this patch.